src/wallet/rpcwallet.cpp

```diff
@@ -3253,25 +3256,27 @@ UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
         return NullUniValue;
     if (fHelp || params.size() < 1 || params.size() > 3)
         throw runtime_error(
-            "z_getaddressforaccount account ( [\"pool\", ...] diversifier_index )\n"
+            "z_getaddressforaccount account ( [\"receiverType\", ...] diversifier_index )\n"
             "\nFor the given account number, derives a Unified Address in accordance"
             "\nwith the remaining arguments:\n"
-            "\n- If no list of pools is given (or the empty list \"[]\"), the best and"
-            "\n  second-best shielded pools, along with the transparent pool, will be used."
+            "\n- If no list of receiver types is given (or the empty list \"[]\"), the best"
+            "\n  and second-best shielded receiver types, along with the transparent receiver"
+            "\n  type, will be used."
             "\n- If no diversifier index is given, the next unused index (that is valid"
-            "\n  for the list of pools) will be selected.\n"
+            "\n  for the list of receiver types) will be selected.\n"
             "\nThe account number must have been previously generated by a call to the"
             "\nz_getnewaccount RPC method.\n"
             "\nOnce a Unified Address has been derived at a specific diversifier index,"
             "\nre-deriving it (via a subsequent call to z_getaddressforaccount with the"
             "\nsame account and index) will produce the same address with the same list"
-            "\nof pools. An error will be returned if a different list of pools is given.\n"
+            "\nof receiver types. An error will be returned if a different list of receiver"
+            "\ntypes is requested.\n"
             "\nResult:\n"
             "{\n"
-            "  \"account\": n,                    (numeric) the specified account number\n"
-            "  \"diversifier_index\": n,          (numeric) the index specified or chosen\n"
-            "  \"pools\": [\"pool\",...]\",         (json array of string) the pools (e.g. \"transparent\", \"orchard\") for which the UA contains receivers\n"
-            "  \"address\"                        (string) The corresponding address\n"
+            "  \"account\": n,                          (numeric) the specified account number\n"
+            "  \"diversifier_index\": n,                (numeric) the index specified or chosen\n"
+            "  \"receiverTypes\": [\"orchard\",...]\",    (json array of string) the receiver types that the UA contains (valid values are \"transparent\", \"sapling\", \"orchard\")\n"
```

Comment:
To match `listaddresses`:
```suggestion
            "  \"receiver_types\": [\"orchard\",...]\",    (json array of string) the receiver types that the UA contains (valid values are \"p2pkh\", \"sapling\", \"orchard\")\n"
```

Also, in #5534 we'd noted at @daira's suggestion that we use `item_types` instead to support inclusion of UFVKs. However, in both `listaddresses` and `z_getaddressforaccount` we are returning information that is specifically about a UA, so I think I prefer `receiver_types` in this case. Thoughts?

Reply:
Agreed, receiver_types is more precise here.